### PR TITLE
Added new feature for read function (issue #9)

### PR DESCRIPTION
**Pull request by EdgarMao**
file chooser pops up when directory is not designated

### CODE REVIEW

**Comment by EdgarMao** [comment]
The first commit had significant problems, the second commit works well.

**Comment by varun-iyer** [comment]
@EdgarMao , what behavior does this have if the user doesn't have an x session running (e.g. over an ssh session)

**Comment by EdgarMao** [comment]
> @EdgarMao , what behavior does this have if the user doesn't have an x session running (e.g. over an ssh session)

I have only ran this on starmaker and my personal computer locally so far, and it works fine in both cases. Since the tkinter library is fairly common on most windows and unix systems, I would imagine it working fine most of the time (I haven't really checked).

**Comment by varun-iyer** [comment]
This is good! Please add a sensible error message if a tkinter window cannot be opened due to ssh/other reason.

**Comment by EdgarMao** [comment]
Hey Varun, in the lastest commit I added the feature where if any error occurs when the file dialog is called, the program will output the following message: "Visual file dialog does not exist, please specify path to directory to save fitsfiles."

File: commands/fitsio.py [hunk 1]
```diff
@@ -8,8 +8,13 @@
 @click.option('-d', '--directory', type=str, help="Specify path to directory of fitsfiles.", required=False)
 @sdi.generator
 def read(directory):
+    # add try (evaluate if the try is efficient)
     if directory == None:
-        paths = glob.glob("{}/*.fits*".format(tkinter.filedialog.askdirectory()))
+        try:
+            paths = glob.glob("{}/*.fits*".format(tkinter.filedialog.askdirectory()))
+        except:
```
Comment (varun-iyer) [inline commands/fitsio.py:15]: This looks good Edgar! However, `except` without specifying a kind of error to except is a bad practice. For example, if the directory given to glob does not exist, glob might throw a FileNotFoundError, which would be caught by `except`, resulting in the very strange error message of having a visual file dialog not exist.

A good fix which does catch all tkinter errors might be to separate out the `tkinter.filedialog.askdirectory()` into its own try-catch, leaving the glob part separate.

File: commands/fitsio.py [hunk 2]
```diff
@@ -8,8 +8,13 @@
...
+            print("Visual file dialog does not exist, please specify path to directory to save fitsfiles.")
```
Comment (varun-iyer) [inline commands/fitsio.py:16]: Use `click.echo` with `err=True` over `print` when possible!

Comment (varun-iyer) [reply commands/fitsio.py:16]: Also, this error message could be a little more specific: e.g., specify path to directory to _read_ fitsfiles with `-d` or something.

**Review by varun-iyer — CHANGES_REQUESTED** [review]

**Comment by EdgarMao** [comment]
Both issues have been resolved, please check the latest commit. Thanks!

**Comment by varun-iyer** [comment]
swallowed + merge in pr #17

**Pull request closed without merging**